bin/src/server.rs

```diff
@@ -76,3 +85,58 @@ pub async fn serve_static(
             .unwrap(),
     }
 }
+
+pub async fn restart_daemon(
+    State(state): State<Arc<ServerState>>,
+) -> Result<(StatusCode, String), (StatusCode, String)> {
+    let mut restart_tx = state.daemon_restart_tx.write().await;
+
+    if let Some(sender) = restart_tx.take() {
+        sender.send(()).map_err(|_| {
+            (
+                StatusCode::INTERNAL_SERVER_ERROR,
+                "couldn't send restart signal".to_string(),
+            )
+        })?;
+        Ok((StatusCode::ACCEPTED, "restart signal sent".to_string()))
+    } else {
+        Ok((
+            StatusCode::ACCEPTED,
+            "restart already triggered".to_string(),
+        ))
+    }
+}
+
+pub async fn get_config(
+    State(state): State<Arc<ServerState>>,
+) -> Result<Json<Config>, (StatusCode, String)> {
+    let config = parse_config(&state.config_path).await.map_err(|err| {
+        (
+            StatusCode::INTERNAL_SERVER_ERROR,
+            format!("failed to read config file: {}", err),
+        )
+    })?;
+
+    Ok(Json(config))
+}
+
+pub async fn set_config(
+    State(state): State<Arc<ServerState>>,
+    Json(config): Json<Config>,
```

Comment:
just to double-check my assumption, since we're saying this is a JSON-serialized `Config` struct, axon handles invalid input such as `config.ui_level = "foo"` right?

Reply:
yes it'll just return a 400 or similar